src/viper/trans.ml

```diff
@@ -532,6 +542,48 @@ and stmt ctxt (s : M.exp) : seqn =
   | _ ->
      unsupported s.at (Arrange.exp s)
 
+(* Translate switch(scrut){...}. *)
+and switch_stmts ctxt at (scrut : M.exp) (cases : M.case list) : seqn' =
+  match cases with
+  | [] -> [], []
+  | c :: cs -> begin
+      let (!!) a = !!! (c.at) a in
+      let {M.pat = p; M.exp = e} = c.it in
+      let conds, p_scope, (p_ds, p_stmts) = pat_match ctxt scrut p in
+      let local_ctxt = add_locals ctxt p_scope in
+      let {it = (e_ds, e_stmts);_} = stmt local_ctxt e in
+      let the = (p_ds @ e_ds, p_stmts @ e_stmts) in
+      let els = switch_stmts ctxt at scrut cs in
+      [], [!!(IfS(conjoin conds at, !!the, !!els))]
```

Comment:
Suggestion: do `cojoin conds` in `pat_match`

Reply:
It makes code a bit more general and maybe a bit simpler. But I am fine with current code.

Reply:
I think it's better to return a list of conditions from `pat_match` so that it's possible to check for irrefutable patterns by `conds == []`